src/main/java/com/blackducksoftware/integration/hub/encryption/PasswordDecrypter.java

```diff
@@ -0,0 +1,167 @@
+package com.blackducksoftware.integration.hub.encryption;
+
+import java.io.InputStream;
+import java.io.PrintWriter;
+import java.io.StringWriter;
+import java.nio.charset.Charset;
+import java.security.InvalidKeyException;
+import java.security.Key;
+import java.security.KeyStore;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+import java.security.UnrecoverableKeyException;
+import java.security.cert.CertificateException;
+import java.util.Arrays;
+
+import javax.crypto.BadPaddingException;
+import javax.crypto.Cipher;
+import javax.crypto.IllegalBlockSizeException;
+import javax.crypto.NoSuchPaddingException;
+
+import org.apache.commons.codec.binary.Base64;
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.slf4j.helpers.NOPLogger;
+
+public final class PasswordDecrypter {
+	private static Logger logger = LoggerFactory.getLogger(PasswordDecrypter.class);
+
+	private static final Charset UTF8 = Charset.forName("UTF-8");
+	private static final String EMBEDDED_SUN_KEY_FILE = "/Sun-Key.jceks";
+	private static final String EMBEDDED_IBM_KEY_FILE = "/IBM-Key.jceks";
```

Comment:
We may want to create a Util class that contains the constants and the shared methods Ex : getKey

We may also want to remove the Logger and pass in an IntLogger, then we can also remove the logError method since that should be handled by the Intlogger